healpy/src/_line_integral_convolution.pyx

```diff
@@ -0,0 +1,124 @@
+import numpy as np
+cimport numpy as np
+cimport cython
+from _common cimport RING, Healpix_Map, ndarray2map
+from pixelfunc import npix2nside
+from healpy.sphtfunc import alm2map, Alm
+
+cdef extern from "_line_integral_convolution_lib.h":
+    cdef void lic_main(const Healpix_Map[double] &Q, const Healpix_Map[double] &U,
+                       const Healpix_Map[double] &th, Healpix_Map[double] &hit,
+                       Healpix_Map[double] &tex, Healpix_Map[double] &mag,
+                       int steps, int kernel_steps, double step_radian,
+                       double polmin, double polmax)
+
+
+def line_integral_convolution(
+    Q,
+    U,
+    steps=100,
+    kernel_steps=50,
+    step_radian=0.003,
+    pol_min=-1e30,
+    pol_max=1e30,
+    rand_seed=42,
+    ell=None,
+    texture=None,
+    modulate=False,
+):
+    """Computes line integral convolution of polarized map.
+
+    Parameters
+    ----------
+    Q : array-like, shape (Npix,)
+      Input Stokes Q map. Must be in RING ordering.
+    U : array-like, shape (Npix,)
+      Input Stokes U map. Must be in RING ordering.
+    steps : int
+      Number of steps to use for each line of the convolution (default: 100).
+    kernel_steps : int
+      Extent of the convolution kernel (in steps, <= steps) (default: 50).
+    step_radian : float
+      Size (in radians) of each step in the convolution (default: 0.003).
+    pol_min : float
+      Minimum value for the polarization magnitude (default: -1e30).
+    pol_max : float
+      Maximum value for the polarization magnitude (default: 1e30).
+    rand_seed : int
+      Seed for the random number generator; only used when texture is not
+      provided (default: 42).
```

Comment:
what about using `None` as default seed?

Reply:
C++ codes does not have the concept of a `None` seed meaning "initialize the RNG in a nonreproducible way".  And I'd argue that this is actually the better choice in a scientific context. In game deveopment, things would probably be different :)

Reply:
I moved the PRNG use into the Cython file, so the C++ restrictions don't apply (of course `None` could just have been interpreted to seed the C++ PRNG with a random number from NumPy's PRNG). The fixed seed was used to ensure reproducible output; the default seed value was copied from the `alice3` executable.